### ISSUE TITLE
[BUG] Integrate CDK fix for Route53 VPC region

### ISSUE THREAD

**Issue by cartalla** [aws-samples/aws-eda-slurm-cluster#46]
**Describe the bug**

I added a custom resource to add regional VPCs to the Route53 hosted zone because of the following CDK bug.

https://github.com/aws/aws-cdk/issues/20496

Need to get the latest version of CDK that has the fix and verify the fix.
The latest version today, 9/7/22, is 2.40.0 and we're currently using version 2.28.1.

**Comment by cartalla** [comment]
Version 2.40.0 is from 7 days ago so it will not have the fix which was merged today.

**Comment by cartalla** [comment]
Log showing 2.41.0 which doesn't have this fix yet.

```
* 46cf825739 "Cory Hall" <43035978+corymhall@users.noreply.github.com> 2 hours ago (HEAD -> main, origin/main)
| fix(elbv2): connections not created for chained listener actions (#21939)
*   ab7668197b "mergify[bot]" <37929162+mergify[bot]@users.noreply.github.com> 14 hours ago
|\  chore(merge-back): 2.41.0 (#21958)
| *   dc8abdfdf4 "mergify[bot]" <37929162+mergify[bot]@users.noreply.github.com> 14 hours ago
| |\  Merge branch 'main' into merge-back/2.41.0
| |/  
|/|   
* | aedc8883bf "Joshua Weber" <57131123+daschaa@users.noreply.github.com> 16 hours ago
| | fix(route53): vpc region in template overridden by stack region (#20530)
| *   6ad48a3193 "mergify[bot]" <37929162+mergify[bot]@users.noreply.github.com> 16 hours ago (tag: v2.41.0, origin/v2-release)
| |\  chore(release): 2.41.0 (#21955)
| | * 28e080aa45 "AWS CDK Team" <aws-cdk@amazon.com> 17 hours ago
| |/  chore(release): 2.41.0
|/|   
* | b0ba52e5a0 "Kendra Neil" <53584728+TheRealAmazonKendra@users.noreply.github.com> 18 hours ago
```

**Comment by cartalla** [comment]
The bug fix was released in 2.42.0.

I tested with 2.42.1 and am still seeing the same problem. The Route53 code is not getting the region from vpc.env.region, but the region is not set correctly by the context provider.

Filed

https://github.com/aws/aws-cdk/issues/22178

**Comment by cartalla** [comment]
Deprecating the legacy scheduler in favor of ParallelCluster.
